src/skeleplex/graph/skeleton_graph.py

```diff
@@ -209,6 +229,52 @@ def orient_splines(graph: nx.DiGraph) -> nx.DiGraph:
     return graph
 
 
+def sample_slices_for_edge(
+    u,
+    v,
+    spline,
+    volume_path,
+    segmentation_path,
+    positions,
+    slice_size,
+    pixel_size,
+    interpolation_order,
+    image_voxel_size_um,
+    approx,
+):
+    """Helper function for parallel execution."""
+    logger.info(f"Sampling slices from edge ({u}, {v})")
+
+    volume = da.from_zarr(volume_path, chunks="auto")
+
+    image_slice = spline.sample_volume_2d(
+        volume,
+        positions,
+        grid_shape=(slice_size, slice_size),
+        grid_spacing=(pixel_size, pixel_size),
+        sample_interpolation_order=interpolation_order,
+        sample_fill_value=0,
+        image_voxel_size_um=image_voxel_size_um,
+        approx=approx,
+    )
+
+    segmentation_slice = None
+    if segmentation_path is not None:
+        segmentation = da.from_zarr(segmentation_path, chunks="auto")
+        segmentation_slice = spline.sample_volume_2d(
+            segmentation,
+            positions,
+            grid_shape=(slice_size, slice_size),
+            grid_spacing=(pixel_size, pixel_size),
+            sample_interpolation_order=0,
+            image_voxel_size_um=image_voxel_size_um,
+            sample_fill_value=0,
+            approx=approx,
+        )
+
+    return (u, v), image_slice, segmentation_slice
```

Comment:
I think it would be cleaner to just have the function accept a single path and then when you need to do the image and segmentation (and other images) just call the function multiple times.